### ISSUE TITLE
SURVIVOR step is not progressing

### ISSUE THREAD

**Issue by KristinaGagalova** [aquaskyline/LRSIM#12]
Hi, 
I am running LRSIM on human chr22 and I am having some troubles with the SURVIVOR step. This is the status file:
```
humanChr22.status exists
Tue Nov  7 15:19:59 2017: humanChr22.status
Tue Nov  7 15:19:59 2017: Variant simulation mode enabled
Tue Nov  7 15:19:59 2017: SURVIVOR start
Tue Nov  7 15:19:59 2017: Running: /home/myname/path/todir/bin/LRSIM/SURVIVOR 0 chr22.fa humanChr22.hap.parameter 0 humanChr22.hap 1000
```
I have tried to run the example independently and it hangs at this point, no error:
```
/home/myname/path/todir/bin/LRSIM/SURVIVOR 0 chr22NoN.fa humanChr22.hap.parameter 0 humanChr22.hap 1000
# Chrs passed size threshold:1
# Chrs passed size threshold:1
First: Genome checking:
First: Genome checking:
generate SV
generate_mutations_diploid function
```
Any suggestions?
Thank you in advance

**Comment by KristinaGagalova** [comment]
I would like to give an update:
I have checked and the input format in SURVIVOR looks fine, so my input data in SURVIVOR are correct. 
The thing that I am noticing is that when I run with default parameters, it hangs at the survivor point and stays there for many hours without giving any error. Command below:
```
perl /home/kgagalova/arch/filer/bin/LRSIM/simulateLinkedReads.pl -r chr22.fa -p humanChr22 -x 400 -t 200 -m 10 -z 12
```
```
Wed Nov  8 11:42:49 2017: humanChr22.status
Wed Nov  8 11:42:49 2017: Variant simulation mode enabled
Wed Nov  8 11:42:49 2017: SURVIVOR start
Wed Nov  8 11:42:49 2017: Running: /home/kgagalova/arch/filer/bin/LRSIM/SURVIVOR 0 chr22.fa humanChr22.hap.parameter 0 humanChr22.hap 1000
```
------
if instead I use other type of parameters it runs without problems:

```
perl /home/kgagalova/arch/filer/bin/LRSIM/simulateLinkedReads.pl -r chr22NoN.fa -p humanChr22 -x 400 -t 200 -m 10 -z 12 -1 10000 -4 0 -7 0 -0 0
```
```
Wed Nov  8 08:53:37 2017: humanChr22.status
Wed Nov  8 08:53:37 2017: Variant simulation mode enabled
Wed Nov  8 08:53:37 2017: SURVIVOR start
Wed Nov  8 08:53:37 2017: Running: /home/kgagalova/arch/filer/bin/LRSIM/SURVIVOR 0 chr22.fa humanChr22.hap.parameter 0 humanChr22.hap 10000
Wed Nov  8 08:53:45 2017: SURVIVOR end
Wed Nov  8 08:53:45 2017: Build genome index start
Wed Nov  8 08:53:45 2017: /home/kgagalova/arch/filer/bin/LRSIM/faFilter.pl humanChr22.hap.0.fasta 0 > humanChr22.hap.0.clean.fasta
Wed Nov  8 08:53:48 2017: /home/kgagalova/arch/filer/bin/LRSIM/faFilter.pl humanChr22.hap.1.fasta 0 > humanChr22.hap.1.clean.fasta
Wed Nov  8 08:53:51 2017: /home/kgagalova/arch/filer/bin/LRSIM/samtools faidx humanChr22.hap.0.clean.fasta
Wed Nov  8 08:53:55 2017: /home/kgagalova/arch/filer/bin/LRSIM/samtools faidx humanChr22.hap.1.clean.fasta
Wed Nov  8 08:53:59 2017: Build genome index end
Wed Nov  8 08:53:59 2017: DWGSIM round 0 thread 0 start
Wed Nov  8 08:53:59 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.0.clean.fasta humanChr22.dwgsim.0.0
Wed Nov  8 08:54:01 2017: DWGSIM round 0 thread 1 start
Wed Nov  8 08:54:01 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.0.clean.fasta humanChr22.dwgsim.0.1
Wed Nov  8 08:54:03 2017: DWGSIM round 0 thread 2 start
Wed Nov  8 08:54:03 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.0.clean.fasta humanChr22.dwgsim.0.2
Wed Nov  8 08:54:06 2017: DWGSIM round 0 thread 3 start
Wed Nov  8 08:54:06 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.0.clean.fasta humanChr22.dwgsim.0.3
Wed Nov  8 08:54:10 2017: DWGSIM round 0 thread 4 start
Wed Nov  8 08:54:10 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.0.clean.fasta humanChr22.dwgsim.0.4
Wed Nov  8 08:54:14 2017: DWGSIM round 0 thread 5 start
Wed Nov  8 08:54:14 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.0.clean.fasta humanChr22.dwgsim.0.5
Wed Nov  8 08:54:17 2017: DWGSIM round 1 thread 0 start
Wed Nov  8 08:54:17 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.1.clean.fasta humanChr22.dwgsim.1.0
Wed Nov  8 08:54:19 2017: DWGSIM round 1 thread 1 start
Wed Nov  8 08:54:19 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.1.clean.fasta humanChr22.dwgsim.1.1
Wed Nov  8 08:54:23 2017: DWGSIM round 1 thread 2 start
Wed Nov  8 08:54:23 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.1.clean.fasta humanChr22.dwgsim.1.2
Wed Nov  8 08:54:25 2017: DWGSIM round 1 thread 3 start
Wed Nov  8 08:54:25 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.1.clean.fasta humanChr22.dwgsim.1.3
Wed Nov  8 08:54:27 2017: DWGSIM round 1 thread 4 start
Wed Nov  8 08:54:27 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.1.clean.fasta humanChr22.dwgsim.1.4
Wed Nov  8 08:54:31 2017: DWGSIM round 1 thread 5 start
Wed Nov  8 08:54:31 2017: /home/kgagalova/arch/filer/bin/LRSIM/dwgsim -N 50000000 -e 0.0001,0.0016 -E 0.0001,0.0016 -d 350 -s 35 -1 135 -2 151 -H -y 0 -S 0 -c 0 -m /dev/null humanChr22.hap.1.clean.fasta humanChr22.dwgsim.1.5
Wed Nov  8 09:56:13 2017: DWGSIM round 0 thread 1 end
Wed Nov  8 09:56:13 2017: DWGSIM round 0 thread 3 end
Wed Nov  8 09:56:18 2017: DWGSIM round 0 thread 0 end
Wed Nov  8 09:56:18 2017: cat humanChr22.dwgsim.0.1.12.fastq >> humanChr22.dwgsim.0.12.fastq
Wed Nov  8 09:56:23 2017: DWGSIM round 0 thread 2 end
Wed Nov  8 09:56:35 2017: DWGSIM round 1 thread 4 end
Wed Nov  8 09:56:39 2017: DWGSIM round 1 thread 5 end
Wed Nov  8 09:56:39 2017: DWGSIM round 1 thread 0 end
...
```
I am not sure if the parsing works properly or the problem comes from SURVIVOR....

**Comment by sjackman** [comment]
This issue may be caused by https://github.com/fritzsedlazeck/SURVIVOR/issues/8
`SURVIVOR stalls when reference FASTA lacks newlines`

**Comment by sjackman** [comment]
This issue may hopefully be resolved by https://github.com/fritzsedlazeck/SURVIVOR/commit/2f5f94e00959fb448215c3e5f925d289cbd980d8
You may want to update your vendored copy of SURVIVOR: https://github.com/aquaskyline/LRSIM/blob/master/SURVIVORSrc/src/simulator/SV_Simulator.cpp

**Comment by aquaskyline** [comment]
Fixed.

**Comment by sjackman** [comment]
For reference, fixed by 3c71dbd8ae100b7331cd09ceb66510b977578074
> fix SURVIVOR stalls when reference FASTA lacks newlines

**Comment by sjackman** [comment]
Thanks, Ruibang!
